### PR TITLE
test: add tests for QuarkusInstantiator

### PR DESCRIPTION
**Pull request by mcollovati**

### CODE REVIEW

File: integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            String uuid = waitUntil(d -> d.findElement(By.id("COMPONENT_ID")))
+                    .getText();
+
+            waitUntil(d -> !prevCounter.get()
+                    .equals(d.findElement(By.id("CLICK_COUNTER")).getText()));
+
+            if (prevUuid != null) {
+                Assert.assertEquals("UUID should not have been changed",
+                        prevUuid, uuid);
+            }
```
Comment (caalador) [inline integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java:30]: As the test actually tests that the initial UUID stays through all navigations, should we instead only initialize the uuid once?
```suggestion
            if (prevUuid != null) {
                Assert.assertEquals("UUID should not have been changed",
                        prevUuid, uuid);
            } else {
                prevUuid = uuid;
            }
```
perhaps rename prevUuid to viewUuid or something more static.

File: runtime/src/test/java/com/vaadin/quarkus/QuarkusInstantiatorTest.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    @AfterEach
+    public void tearDown() {
+        serviceUnderTestContext.tearDownAll();
+    }
+
+    @Test
+    public void getI18NProvider_beanEnabled_instanceReturned() {
+        I18NProvider i18NProvider = instantiator.getI18NProvider();
+        Assertions.assertNotNull(i18NProvider);
+        Assertions.assertTrue((i18NProvider instanceof I18NTestProvider));
```
Comment (caalador) [inline runtime/src/test/java/com/vaadin/quarkus/QuarkusInstantiatorTest.java:166]:
```suggestion
        Assertions.assertTrue(i18NProvider instanceof I18NTestProvider);
```
nit: some extra parentheses

File: integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java (hunk 1 above)
Comment (mcollovati) [reply integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java:30]: Good catch! I'll fix it also on Flow test

Comment (mcollovati) [reply integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java]: done

**Review by caalador — APPROVED** [review]

**Pull request merged**